### PR TITLE
Closes #2546 Updating ABLFL to Null if not "Y" in `derive_var_base()`example

### LINKED ISSUE #2546: Documentation:  Update the example in the vignette derive_var_base to have only “Y” or null values for ABLFL.
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

The example contains “N” for ABLFL, but ABLFL should only contain “Y” and null values. Therefore, I suggest updating the example in the vignette to include only “Y” and null values for ABLFL.

image

Codelist term for ABLFL in ADaM IG 1.3

image

### PR DESCRIPTION
**Pull request by sadchla-codes**
… example

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi @sadchla-codes is this ready for review?

**Review by bms63 — COMMENTED** [review]
Can you check to see if other examples have ABLFL = N please

**Comment by sadchla-codes** [comment]
> Can you check to see if other examples have ABLFL = N please

Question @bms63: Would you prefer ABLFL ="N" be replace with blank ("") or NA_character_?

**Comment by bundfussr** [comment]
> > Can you check to see if other examples have ABLFL = N please
> 
> Question @bms63: Would you prefer ABLFL ="N" be replace with blank ("") or NA_character_?

I think our convention is to use `NA_character_`.

**Review by bms63 — COMMENTED** [review]
oooohhhhh - standardizing!!  @cpiraux look what you have wrought :)

**Review by bms63 — COMMENTED** [review]
I think this update warrants an entry in the news now!

For character values can you manually left aligned and numeric values right aligned where possible?  This will increase readability for the examples and tests when others are reviewing.  

I believe you can make this manual update and styler won't complain.  Is that correct @bundfussr ?

**Comment by bms63** [comment]
@sadchla-codes can you please align the test data so easier for folks to read the data sets in the examples and test data?  tyty

**Comment by bms63** [comment]
Here is that weird one with styler @bundfussr 

Manual setting:
image

Running styler resets it - the one above is unchanged but was manually set. 
image

If you don't see anything - I say we just move on and get this merged in!

**Comment by bundfussr** [comment]
> Here is that weird one with styler @bundfussr
> 
> Manual setting: image
> 
> Running styler resets it - the one above is unchanged but was manually set. image
> 
> If you don't see anything - I say we just move on and get this merged in!

The alignment must include the header. For `expected_output` `~BASETYPE` is not aligned to the contents of the column.

**Comment by bms63** [comment]
got it - what an art 🥇 

image

**Comment by bms63** [comment]
@bundfussr can i get your blessing/sanity check on this one we have a moment.

File: R/derive_var_base.R [hunk 1]
```diff
@@ -43,19 +43,19 @@
...
+#'   "TEST01", "PAT01", "PARAM01", 15.01, NA, "Day 14", NA_character_, "HIGH",
+#'   "TEST01", "PAT01", "PARAM02", 8.350, NA, "Baseline", "Y", "LOW",
+#'   "TEST01", "PAT01", "PARAM02", NA, NA, "Day 7", NA_character_, NA,
+#'   "TEST01", "PAT01", "PARAM02", 8.350, NA, "Day 14", NA_character_, "LOW",
+#'   "TEST01", "PAT01", "PARAM03", NA, "LOW", "Baseline", "Y", NA,
+#'   "TEST01", "PAT01", "PARAM03", NA, "LOW", "Day 7", NA_character_, NA,
+#'   "TEST01", "PAT01", "PARAM03", NA, "MEDIUM", "Day 14", NA_character_, NA,
+#'   "TEST01", "PAT01", "PARAM04", NA, "HIGH", "Baseline", "Y", NA,
+#'   "TEST01", "PAT01", "PARAM04", NA, "HIGH", "Day 7", NA_character_, NA,
+#'   "TEST01", "PAT01", "PARAM04", NA, "MEDIUM", "Day 14", NA_character_, NA
```
[the remark this answers is not in the document]
Comment (bms63) [reply R/derive_var_base.R:58]: oh i did not know that

[the remark this answers is not in the document]
Comment (bms63) [reply R/derive_var_base.R]: aligned

File: vignettes/generic.Rmd [hunk 2]
```diff
@@ -208,12 +208,12 @@ adsl <- tribble(
...
-  "1",      "HEIGHT", "BASELINE", "Y",    169.2, "cm",
-  "1",      "WEIGHT", "WEEK 3",   NA,      59.3, "kg",
-  "2",      "WEIGHT", "BASELINE", "Y",     72.5, "kg",
-  "2",      "WEIGHT", "WEKK 3",   NA,      71.9, "kg",
+  ~USUBJID, ~PARAMCD, ~AVISIT, ~ABLFL, ~AVAL, ~AVALU,
+  "1", "WEIGHT", "BASELINE", "Y", 58.7, "kg",
+  "1", "HEIGHT", "BASELINE", "Y", 169.2, "cm",
+  "1", "WEIGHT", "WEEK 3", NA_character_, 59.3, "kg",
+  "2", "WEIGHT", "BASELINE", "Y", 72.5, "kg",
+  "2", "WEIGHT", "WEKK 3", NA_character_, 71.9, "kg",
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/generic.Rmd]: aligned

**Comment by bms63** [comment]
> The examples in `derive_var_chg()` and `derive_var_pchg()` should be updated.

updated

**Review by bundfussr — APPROVED** [review]
